### PR TITLE
Remove fallback for controller location

### PR DESCRIPTION
**Pull request by droberts195**
This change removes the temporary controller
location fallback introduced in #47013.

Relates elastic/ml-cpp#593

### CODE REVIEW

**Comment by droberts195** [comment]
Important: do not merge this until a few days after elastic/ml-cpp#593 is merged, otherwise it will disrupt a lot of developers working on Macs.

**Comment by droberts195** [comment]
> (The PR CI is green because it ran on Linux. It would have failed on Mac.)

Since elastic/ml-cpp#593 was merged 3 days ago this is no longer the case.  I have run the ML native multi-node tests locally on a Mac and they passed.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**
